### ISSUE TITLE
Ability to download .HEX from Arduino [feature request]

### ISSUE THREAD

**Issue by DRSDavidSoft** [xxxajk/PhysicaloidLibrary#5]
Physicaloid currently can "upload" (write) .hex files, i.e. burning firmware to the chip.
However, the STK500 protocol also has the ability to "download" (read) flash content of the chip.

Does [the current implementation](https://github.com/xxxajk/PhysicaloidLibrary/tree/master/PhysicaloidLibrary/src/com/physicaloid/lib/programmer/avr) have this feature? If not, could you please port it from [the official AVRdude](https://svn.savannah.gnu.org/viewvc/avrdude/trunk/avrdude/) repo?

**Comment by xxxajk** [comment]
No, and while I could do that, I would have to find the time to do so.
It was a lucky coincidence that I had a few free days to implement and test the last patch set.
Perhaps it is something you could do, in order to learn how to do cross-language porting.
Java is VERY close to C/C++, you just have to know the subtle differences.

**Comment by xxxajk** [comment]
Actually, if you look closely at the current source, it already contains much of what you would need.
All you would have to do is make a few extra classes in order to do it.
